app/controllers/api/authentications_controller.rb

```diff
@@ -1,4 +1,20 @@
 module Api
   class AuthenticationsController < BaseController
+    def create_resource(_type, _id, data)
+      validate_auth_attrs(data)
+      klass = data['type'].constantize
```

Comment:
we shouldn't be constantizing based on user specified data. Let's have the model lookup data['type'] for us and return the appropriate klass.